tests/bsim/bluetooth/mesh/tests_scripts/op_agg/model_coex.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+# Copyright 2023 Nordic Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+source $(dirname "${BASH_SOURCE[0]}")/../../_mesh_test.sh
+
+# Test that Opcode aggregator server and client can coexist on the same device.
+# In this test scenario, the DUT has a sequence in progress on the Opcode
+# Aggregator client model that is interrupted by an incoming sequence to the Opcode
+# Aggregator server model. The test verifies that both the incoming and outgoing
+# sequence on the DUT completes successfully.
+#
+# Test procedure:
+# 1. Both Tester and DUT device initialize both Opcode aggregator server and client.
+#    Both devices starts an Opcode aggregator sequence and populates the buffer.
+# 2. The Tester device immediately starts sending the sequence.
+# 3. The DUT device waits, and verifies that the sequence is correctly received.
+#    Then it starts sending its own aggregated sequence.
```

Comment:
If I'm not mistaken, the only limitation we have now is that we can't start a new sequence to a device which we are currently processing the incoming sequence from, right? Could this be handled in API?

Reply:
I think it can be started, but as long as the server is in progress it won't be possible to add messages to the cli buffer. In theory, aggregating a message for the client while the server is processing will cause undesired/erroneous behavior since the message will end up in the server response. I think this can happen only if the BT_RX thread is preempted while processing the incoming sequence to the server, but not 100% sure.  Another potential scenario is if one of the receiving models has a side effect of issuing additional messages other than the status responses, but can't think of any particular model (maybe time models?).

>Could this be handled in API?
Could you elaborate what you mean by this?

Reply:
It boils down to the ideal fix of tagging outgoing messages. I think that will help us solve any particular corner case for OpAgg.

Reply:
Ah, I see then it is only the case for loopback on the same device on the same element?

> Could you elaborate what you mean by this?

I though that starting a new sequence should be prevented if incoming sequence is still processed and `if-else` in `bt_mesh_model_send` may cause to put the new sequence messages into the servers response.

Reply:
> Ah, I see then it is only the case for loopback on the same device on the same element?

I think that you could start a sequence at any time without an issue, also in the middle of a server handling a incoming sequence. `bt_mesh_op_agg_cli_seq_start` only initializes its internal buffer and sets the init flag. The init flag in turn allows `bt_mesh_model_send` to put messages into the cli buffer, but this will of course never happen as long as we have the `else if` and the server is initialized.

Reply:
> It will allow it if the recv ctx of the server and dst ctx of the client is not identical.

Is there any check for this in API when starting a new sequence?

Reply:
If srv ctx and cli ctx are identical? No. 

I'm not sure it would help either. A new sequence for the client can be started and be ongoing indefinetly in theory. Within that time window a server might have had multiple different recv ctx.